src/main/java/com/tabnine/binary/fetch/BinaryVersionFetcher.java

```diff
@@ -27,6 +28,12 @@ public BinaryVersionFetcher(LocalBinaryVersions localBinaryVersions, BinaryRemot
      * @throws SecurityException, NoExistingBinaryException if something went wrong
      */
     public String fetchBinary() throws NoValidBinaryToRunException {
+        Optional<BinaryVersion> bootstrappedVersion = BootstrapperSupport.bootstrap_version(localBinaryVersions, binaryRemoteSource, binaryDownloader);
+        if (bootstrappedVersion.isPresent()) {
+            Logger.getInstance(getClass()).info(format("found local bootstrapped version %s", bootstrappedVersion.get()));
+            return bootstrappedVersion.get().getVersionFullPath();
+        }
+        Logger.getInstance(getClass()).warn("couldn't get bootstrapped version. fallback to legacy code!");
```

Comment:
Warning is a bit harsh here don't you think?W

Reply:
the server should always return a version unless problems